test/test_trainer.py

```diff
@@ -155,6 +168,24 @@ def test_classification_trainer_evaluate(self):
         # More than 5 classes
         self._test_classification_trainer(3)
 
+    def test_binary_classification_trainer(self):
+
+        num_it = 10
+        batch_size = 8
+        # Generate all dependencies
+        model = nn.Sequential(nn.Conv2d(3, 32, 3), nn.ReLU(inplace=True),
+                              GlobalAvgPool2d(flatten=True), nn.Linear(32, 1))
+        train_loader = torch.utils.data.DataLoader(MockBinaryClassificationDataset(num_it * batch_size),
+                                                   batch_size=batch_size)
+        optimizer = torch.optim.Adam(model.parameters())
+        criterion = torch.nn.BCELoss()
```

Comment:
We might have to make a proper test, because this wouldn't converge properly : the BCELoss assumes a prediction in the range [0, 1] during training. Considering the architecture in use, this isn't the case

I'd recommend switching to BCEWithLogitsLoss https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html